src/twisted/internet/interfaces.py

```diff
@@ -55,6 +55,70 @@ def getDestination():
 
 
 
+class INameResolver(Interface):
+    """
+    XXX Write me.
+
+    RFC 3484.
+    """
+    def getAddressInformation(name, service, family=None, socktype=None,
+                              proto=None, flags=None, timeout=None):
+        """
+        Get the address information associated with the given name.
+
+        @param name: A hostname to resolve.
+        @type name: C{str}
+
+        @param service: A port number or the name of the service for which to
+            find address information.  For example, C{22} or C{"ssh"}.
+        @type service: C{int} or C{str}
+
+        @param family: If specified, limit results to addresses from this
+            family.  Must be one of the address family constants from the
+            socket module.  For example, L{socket.AF_INET}.
+
+        @param socktype: If specified, limit results to addresses for this
+            socket type.  Must be one of the socket type constants from the
+            socket module.  For example, L{socket.SOCK_STREAM}.
+
+        @param proto: If specified, limit results to addresses for this socket
+            protocol.  Must be one of the protocol constants from the socket
+            module.  For example, L{socket.IPPROTO_TCP}.
+
+        @param flags: A bitvector specifying zero or more of the following::
+
+                - Yea right.  Go read the `getaddrinfo(3)` man page.
+
+        @param timeout: The number of seconds after which to cancel the
+            L{Deferred} returned by this function.  Can be a single C{int} or
+            an iterable of C{int} for implementations which support retries, in
+            which case the L{Deferred} cancellation timeout will be the sum of
+            all the C{int}s.
+        @type: timeout: C{int} or an iterable of C{int}
+
+        @raise ValueError: If one of the specified flags is not supported by
+            the implementation.  All flags are optional.
+
+        @return: A L{Deferred} which will fire when the resolution completes.
+            If resolution is successful, the result will be a list of objects
+            with the following attributes:
```

Comment:
This "with the following attributes" section should really be a reference to a type, rather than an ad-hoc description of attributes.